File: data/go_fetch/api/exchange_flows.go
package api

import (
	"fmt"
	"io"
	"net/http"

	"github.com/sonh/qs"
)

func Reserve() string {
	url := "https://api.datasource.cybotrade.rs/cryptoquant/btc/exchange-flows/reserve"

	req, _ := http.NewRequest("GET", url, nil)
	req.Header.Add("accept", "application/json")
	req.Header.Add("X-API-KEY", "iheM86n8mn8vC3vjOE444vlVTP5sTuBb71FJuVKQc5UqdIBn")

	type Query struct {
		Exchange string `qs:"exchange"`
		Window   string `qs:"window"`
		// From     string `qs:"from"`
		// To       string `qs:"to"`
		// Start_time int64  `qs:"start_time,comma"`
		// End_time   int64  `qs:"end_time"`
		Limit int64 `qs:"limit"`
		// Flatten    bool   `qs:"flatten"`
	}

	query := &Query{
		Exchange: "all_exchange",
		Window:   "day",
		// From:     "20191003",
		// To:       "20191004T220000",
		// Start_time: 1735689600,
		// End_time:   1738368000,
		Limit: 20,
		// Flatten:    true,
	}

	encoder := qs.NewEncoder()
	values, err := encoder.Values(query)
	if err != nil {
		// Handle error
		fmt.Println(err)
	}

	req.URL.RawQuery = values.Encode()

	res, err := http.DefaultClient.Do(req)
	if err != nil {
		println(err)
	}

	defer res.Body.Close()

	body, _ := io.ReadAll(res.Body)

	return string(body)
}

File: data/go_fetch/api/exchange_flows.py
import requests
import json
from urllib.parse import urlencode

def reserve():
    url = "https://api.datasource.cybotrade.rs/cryptoquant/btc/exchange-flows/reserve"
    
    headers = {
        "accept": "application/json",
        "X-API-KEY": "iheM86n8mn8vC3vjOE444vlVTP5sTuBb71FJuVKQc5UqdIBn"
    }
    
    query_params = {
        "exchange": "all_exchange",
        "window": "day",
        "limit": 20
    }
    
    # Build the URL with query parameters
    full_url = f"{url}?{urlencode(query_params)}"
    
    try:
        # Make the GET request
        response = requests.get(full_url, headers=headers)
        response.raise_for_status()  # Raise an exception for bad status codes
        
        # Parse the JSON response and return the dictionary
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error making request: {e}")
        return None
    except json.JSONDecodeError as e:
        print(f"Error parsing JSON response: {e}")
        return None

if __name__ == "__main__":
    result = reserve()
    if result:
        print(json.dumps(result, indent=2)) 